command/agent/alloc_endpoint.go

```diff
@@ -152,13 +152,23 @@ func (s *HTTPServer) allocStop(allocID string, resp http.ResponseWriter, req *ht
 		var err error
 		noShutdownDelay, err = strconv.ParseBool(noShutdownDelayQS)
 		if err != nil {
-			return nil, fmt.Errorf("no_shutdown_delay value is not a boolean: %v", err)
+			return nil, fmt.Errorf("no_shutdown_delay value is not a boolean: %v", err)
+		}
+	}
+
+	reschedule := false
+	if rescheduleQS := req.URL.Query().Get("reschedule"); rescheduleQS != "" {
+		var err error
+		reschedule, err = strconv.ParseBool(rescheduleQS)
+		if err != nil {
+			return nil, fmt.Errorf("reschedule value is not a boolean: %w", err)
 		}
 	}
```

Comment:
There's a `parseBool` helper in `command/agent/http.go` that you can use like 

```go
reschedule, err := parseBool(req, "reschedule")
if err != nil {
	return nil, err
}
```

(which we could use above for `no_shutdown_delay` as well)